lib/methods/import.js

```diff
@@ -232,12 +236,55 @@ module.exports = self => {
       return locale;
     },
 
-    async insertDocs(req, docs, reporting) {
+    async checkDuplicates(req, docs) {
       const duplicatedDocs = [];
-      const duplicatedIds = [];
+
+      const docIds = docs
+        .filter(({ aposDocId }) => aposDocId)
+        .map(({ aposDocId }) => aposDocId);
+
+      const aposLocale = `${req.locale}:draft`;
+      const criteria = {
+        $and: [
+          { aposDocId: { $in: docIds } },
+          {
+            $or: [
+              { aposLocale: { $exists: false } },
+              { aposLocale }
+            ]
+          }
+        ]
+      };
+
+      const duplicates = await self.apos.doc.db.distinct('aposDocId', criteria);
+      const duplicatedIds = new Set(duplicates);
+
+      for (const doc of docs) {
+        if (duplicatedIds.has(doc.aposDocId) && (!doc.aposMode || doc.aposMode === 'draft')) {
+          duplicatedDocs.push({
+            aposDocId: doc.aposDocId,
+            title: doc.title,
+            type: doc.type,
+            updatedAt: doc.updatedAt
+          });
+        }
+      }
+
+      return {
+        duplicatedIds,
+        duplicatedDocs
+      };
+    },
+
+    async insertDocs(req, {
+      docs, reporting, duplicatedIds
```

Comment:
```suggestion
      docs, reporting, duplicatedAposIds
```

Reply:
Not sure it's  a needed change since we know we use `aposDocId` for imports but we can.